core/shared/src/main/scala/fs2/async/async.scala

```diff
@@ -114,8 +111,8 @@ package object async {
     * nowhere.
    */
   def race[F[_]: Effect, A, B](fa: F[A], fb: F[B])(implicit ec: ExecutionContext): F[Either[A, B]] =
```

Comment:
Perhaps we should move this `race` to `Promise` companion (we can leave the forwarder here though). We should define as `def raceWith[F[_]: Effect, A, B](fa: F[A], fb: F[B])(finished: Either[(A,F[B]),(F[A],B)] => F[Unit]): F[Either[A,B]]` and then define `race` in terms of `raceWith`.

Reply:
Or actually:

```scala
def raceWith[F[_]: Effect, A, B, C](fa: F[A], fb: F[B])(finished: Either[(A,F[B]), (F[A],B)] => F[C]): F[C]
```

Reply:
I'm leaning towards leaving `race` as is in this PR and tackle it in a separate one, since it's mostly self-contained